runners/apex/src/main/java/org/apache/beam/runners/apex/translators/functions/ApexGroupByKeyOperator.java

```diff
@@ -230,7 +232,7 @@ private void processElement(WindowedValue<KV<K, V>> windowedValue) throws Except
     }
     StateInternals<K> stateInternals = perKeyStateInternals.get(keyBytes);
     if (stateInternals == null) {
-      stateInternals = InMemoryStateInternals.forKey(key);
+      stateInternals = stateInternalsFactory.stateInternalsForKey(key);
```

Comment:
This is definitely the right place to call this, so presumably having it transient would be fine? But also fine to leave it as-is, because the logic is fine. Just foolproof to make sure you don't accidentally ship up the `StateInternals` with the operator.